src/frontends/ir/src/ir_deserializer.cpp

```diff
@@ -433,11 +436,15 @@ std::shared_ptr<ngraph::Function> XmlDeserializer::parse_function(
         if (node_param.type == "Result" || node_param.type == "Assign") {
             outputs.push_back(node_param.layerId);
         }
+        if (node_param.type == "Parameter") {
+            // Save Parameters order according to order in XML.
+            // To do so, handle nodes manually and ignore during DFS
+            dfs_used_nodes.insert(node_param.layerId);
+            order.push_back(node_param.layerId);
+            edges[node_param.layerId] = {};
```

Comment:
Why do you need this line?

Reply:
@ilyachur empty values for such layer ids are implicitly added during DFS in https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/ir/src/ir_deserializer.cpp#L457 (when do `edges[id]`). In case when `edges` doesn't contain all ids, it will fall to https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/ir/src/ir_deserializer.cpp#L475 and doesn't fill `id_to_node` in https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/ir/src/ir_deserializer.cpp#L491. That will lead to failure in https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/ir/src/ir_deserializer.cpp#L480 later